src/test/java/com/purbon/kafka/topology/AccessControlManagerTest.java

```diff
@@ -433,222 +398,117 @@ public void newKafkaConnectACLsCreation() {
 
     Topic topicA = new TopicImpl("topicA");
-    project.addTopic(topicA);
-
-    List<Consumer> users = asList(new Consumer("User:app1"));
+    TestTopologyBuilder builder =
+        TestTopologyBuilder.createProject("foo", "project")
+            .addTopic(topicA)
+            .addConsumer("User:app1");
+    List<Consumer> users = new ArrayList<>(builder.getConsumers());
 
     doReturn(singletonList(new TopologyAclBinding()))
         .when(aclsBuilder)
         .buildBindingsForConsumers(users, topicA.toString(), false);
 
-    accessControlManager.apply(topology, plan);
+    accessControlManager.apply(builder.buildTopology(), plan);
 
     plan.run(true);
 
     verify(mockPrintStream, times(1)).println(any(Action.class));
   }
 
   @Test
-  public void testAclDeleteLogic() throws IOException {
-
-    BackendController backendController = new BackendController();
-    backendController.load();
-    backendController.reset();
-
-    Properties props = new Properties();
-    props.put(KAFKA_INTERNAL_TOPIC_PREFIXES, Arrays.asList("foo.", "_"));
-
-    HashMap<String, String> cliOps = new HashMap<>();
-    cliOps.put(BROKERS_OPTION, "");
-    cliOps.put(ALLOW_DELETE_OPTION, "true");
-
-    TopologyBuilderConfig config = new TopologyBuilderConfig(cliOps, props);
-
-    plan = ExecutionPlan.init(backendController, mockPrintStream);
-    accessControlManager = new AccessControlManager(aclsProvider, aclsBuilder, config);
-
-    List<Consumer> consumers = new ArrayList<>();
-    consumers.add(new Consumer("User:app1"));
-    consumers.add(new Consumer("User:app2"));
-
-    Topic topicA = new TopicImpl("topicA");
-
-    Topology topology = buildTopology(consumers, asList(topicA));
-
-    List<TopologyAclBinding> bindings = returnAclsForConsumers(consumers, topicA.getName());
-    doReturn(bindings)
-        .when(aclsBuilder)
-        .buildBindingsForConsumers(any(), eq(topicA.toString()), eq(false));
-
-    accessControlManager.apply(topology, plan);
-    plan.run();
-
-    verify(aclsBuilder, times(1))
-        .buildBindingsForConsumers(refEq(consumers), eq(topicA.toString()), eq(false));
-
-    consumers = new ArrayList<>();
-    consumers.add(new Consumer("User:app1"));
-
-    bindings = returnAclsForConsumers(consumers, topicA.getName());
-    doReturn(bindings)
-        .when(aclsBuilder)
-        .buildBindingsForConsumers(any(), eq(topicA.toString()), eq(false));
-
-    Topology newTopology = buildTopology(consumers, asList(topicA));
-
-    accessControlManager.apply(newTopology, plan);
-    plan.run();
-
-    List<TopologyAclBinding> bindingsToDelete =
-        returnAclsForConsumers(asList(new Consumer("User:app2")), topicA.getName());
+  public void testAclDeleteWithDeleteAllEnabled() throws IOException {
+    doReturn(true).when(config).allowDelete();
 
-    verify(aclsProvider, times(1)).clearBindings(new HashSet<>(bindingsToDelete));
+    verifyAclsDelete();
```

Comment:
I know why you propose this method, as is basically run from two different places, but could we name it differently? The fact that is named _verifyAclsDelete_ seems as is only doing the assert part and not actually running the full test. 

I would prefer a more clear naming or a separation between run and verification of the assertion, otherwise the test is still hard to follow.

Reply:
I was not happy with the name either, but the best I could do at the moment. :-)

But new day, new possibilities! 
- What about testAclsDelete? Because it is actually a full test on acl deletion. Only config is set different.
- Or performAclsDeleteTest?
- Or any suggestion you might have...

Reply:
Who does their best, can not be ask more! this is a Catalan saying! You know I appreciate a ton your contributions.

I think I like most

> What about testAclsDelete? Because it is actually a full test on acl deletion. Only config is set different.

as you said, only config is different.

Reply:
:-) No offence taken @purbon . And a good saying that is.